implants/imix/build.rs

```diff
@@ -1,4 +1,259 @@
+    /// Default: `5`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    interval: Option<u32>,
+
+    /// Duration to wait before retrying this callback if an error occurs, in seconds
+    /// Default: `5`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    retry_interval: Option<u32>,
+
+    /// Override system settings for proxy URI over HTTP(S)
+    /// Only supported for http1 and grpc transports
+    /// (must specify a scheme, e.g. `https://`)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    proxy_uri: Option<String>,
+
+    /// DNS-over-HTTPS provider for gRPC transport
+    /// Valid values: "cloudflare", "google", "quad9"
+    /// If not specified, system DNS resolver will be used
+    /// Only supported for grpc transport
+    #[serde(skip_serializing_if = "Option::is_none")]
+    doh_provider: Option<String>,
+}
+
+/// Build configuration structure matching the environment variables
+/// documented in the user guide
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+struct ImixBuildConfig {
+    /// List of callback configurations
+    /// Each callback can have its own URI, interval, retry_interval, and proxy_uri
+    #[serde(skip_serializing_if = "Option::is_none")]
+    callbacks: Option<Vec<CallbackConfig>>,
+
+    /// The public key for the tavern server
+    /// (obtain from server using `curl $IMIX_CALLBACK_URI/status`)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    server_pubkey: Option<String>,
+
+    /// Manually specify the host ID for this beacon
+    #[serde(skip_serializing_if = "Option::is_none")]
+    host_id: Option<String>,
+
+    /// Imix will only do one callback and execution of queued tasks
+    /// Default: false
+    #[serde(skip_serializing_if = "Option::is_none")]
+    run_once: Option<bool>,
+
+    /// Feature flags for conditional compilation
+    /// Valid values: grpc, http1, dns, win_service
+    /// Note: DoH is now configured per-callback via doh_provider field
+    #[serde(skip_serializing_if = "Option::is_none")]
+    features: Option<Vec<String>>,
+}
+
+/// Parse YAML configuration from string content
+fn parse_yaml_build_config(
+    yaml_content: &str,
+) -> Result<ImixBuildConfig, Box<dyn std::error::Error>> {
+    let config: ImixBuildConfig = serde_yaml::from_str(yaml_content)?;
+    Ok(config)
+}
+
+/// Apply build configuration by setting cargo environment variables
+fn apply_build_config(config: &ImixBuildConfig) {
+    // Handle callbacks
+    if let Some(ref callbacks) = config.callbacks {
+        if !callbacks.is_empty() {
+            // Validate proxy_uri and doh_provider are only set for supported transports
+            for callback in callbacks {
+                let uri_lower = callback.uri.to_lowercase();
+
+                // Validate proxy_uri
+                if let Some(ref proxy_uri) = callback.proxy_uri {
+                    if uri_lower.starts_with("dns://") {
```

Comment:
Change this to a negative match for know supported types http1 and https/http(grpc) this will make it easier to maintain.